channeldb/control_tower.go

```diff
@@ -288,6 +353,16 @@ func (p *paymentControl) Fail(paymentHash lntypes.Hash,
 		return err
 	}
 
+	if updateErr == nil {
+		// Notify subscribers of fail event.
+		p.notifyFinalEvent(
```

Comment:
So there can only be a single subscriber per payment? Consider the following case: two RPC clients race to create a payment only one of them succeeds. IMO, in this scenario, the second should _still_ gain a subscription internally to be notified once the final payment has completed.

Reply:
`notifyFinalEvent` notifies a list of subscribers (multiple)